### DISCUSSION TITLE
Synchronizing Carts Across Mobile and Web

### DISCUSSION THREAD

**Discussion by abdul-tycheros** [medusajs/medusa#9970] [Feature Requests]
To ensure seamless cart synchronization between mobile and web applications, we need to obtain the cart_id as part of the logged-in user details. This allows users to retain their cart contents regardless of the device they use.

Can we get user last **cart_id** along with user details after login in new device?

**Comment by srindom** [comment]
While there is no out-of-the-box support for what you mention it would be pretty simple to support with a custom authenticated endpoint.

```
GET /store/customers/me/carts
# fetch active carts with customer_id = authenticated customer id
```

- When a customer logs in your storefront can make a request to your custom endpoint to check if there are other active carts for the customer.
- Your storefront implementation would have to deal with the returned carts to fit your intended behavior.
